### PR TITLE
Improve Silabs Matter shell functionality/stability

### PR DESCRIPTION
**Pull request by CuRahman**
Issue: Matter shell was experiencing varying functionality across platforms, rendering it unusable on some. The shell did not correctly account for different return cases that could be entered by the user depending on the OS they were running it on. 
The shell also did not account for copy/paste.

- These changes were introduced to allow the matter shell within Silabs projects to allow for usability on Mac, linux, and Windows operating systems.
- Allows for functionality within Simplicity Studio console 
- Allows functionality for copy/paste by introducing a loop to read in characters to the buffer until it cannot
- Improves general stability across platforms

### CODE REVIEW

**Review by Sarthak-Shaha — APPROVED** [review]

File: src/lib/shell/MainLoopSilabs.cpp [hunk 1]
```diff
@@ -131,6 +159,7 @@ int TokenizeLine(char * buffer, char ** tokens, int max_tokens)
     tokens[cursor++] = &buffer[i];
 
     for (; i < len && cursor < max_tokens; i++)
+
```
Comment (mkardous-silabs) [inline src/lib/shell/MainLoopSilabs.cpp:162]:
```suggestion
```

**Review by mkardous-silabs — APPROVED** [review]

File: src/lib/shell/MainLoopSilabs.cpp [hunk 2]
```diff
@@ -32,15 +32,19 @@ using chip::Shell::streamer_get;
 
 namespace {
 
-constexpr const char kShellPrompt[] = "matterCli > ";
+constexpr const char kShellPrompt[] = "matterCli> ";
 
 // max > 1
+bool haveCR = false;
```
Comment (selissia) [inline src/lib/shell/MainLoopSilabs.cpp:38]: Why is this a global?

**Review by selissia — APPROVED** [review]

File: src/lib/shell/MainLoopSilabs.cpp (hunk 2 above)
Comment (CuRahman) [reply src/lib/shell/MainLoopSilabs.cpp:38]: This keeps track of Windows line feeds after ReadLine() has exitted and is ran again

Comment (CuRahman) [reply src/lib/shell/MainLoopSilabs.cpp:38]: This keeps track of Windows line feeds after ReadLine() has exitted and is ran again

**Review by andy31415 — APPROVED** [review]

**Pull request merged**